### PR TITLE
Resolve build warnings

### PR DESCRIPTION
**Pull request by john-s-morgan**
Resolve `obsolete` build warnings.

### CODE REVIEW

File: WalletConnectSharp.Core/WalletConnectSession.cs [hunk 1]
```diff
@@ -580,33 +580,34 @@ protected void HandleSessionDisconnect(string msg, string topic = "disconnect",
 
         if (TransportConnected)
         {
-            DisconnectTransport();
+            // TODO: this is a hack to get the transport to disconnect syncronously.
+            Task.WhenAll(DisconnectTransport());
```
Comment (john-s-morgan) [inline WalletConnectSharp.Core/WalletConnectSession.cs:584]: Realistically should make all the code that depends on this able to `await` this call.

File: WalletConnectSharp.Core/Network/AESCipher.cs [hunk 2]
```diff
@@ -64,9 +64,9 @@ public async Task<string> DecryptWithKey(byte[] key, EncryptedPayload encryptedD
         if (encoding == null)
             encoding = Encoding.UTF8;
 
-        byte[] rawData = encryptedData.data.FromHex();
-        byte[] iv = encryptedData.iv.FromHex();
-        byte[] hmacReceived = encryptedData.hmac.FromHex();
+        byte[] rawData = encryptedData.data.HexToByteArray();
+        byte[] iv = encryptedData.iv.HexToByteArray();
+        byte[] hmacReceived = encryptedData.hmac.HexToByteArray();
```
Comment (AddressXception) [inline WalletConnectSharp.Core/Network/AESCipher.cs:69]: whoops. i also fixed this as part of #49 but it should be a straightforward rebase

File: WalletConnectSharp.Core/Utils/Hex.cs [hunk 3]
```diff
@@ -144,7 +144,7 @@ private static byte FromCharacterToByte(char character, int index, int shift = 0
         return value;
     }
 
-    [Obsolete]
+    [Obsolete("use the builtin string.HexToByteArray. This method will be removed in a future release.")]
```
Comment (john-s-morgan) [inline WalletConnectSharp.Core/Utils/Hex.cs:147]: Good call out from #49. Thanks for the contribution @AddressXception.

**Comment by john-s-morgan** [comment]
Most of the good updates are in #49.

**Pull request closed without merging**